### ISSUE TITLE
[BUG] On mobile-webkit devices, Page keeps resizing by itself and stop being responsive. 

### ISSUE THREAD

**Issue by Qibaoooo** [microsoft/playwright#17152]
**Context:**
- Playwright Version: mcr.microsoft.com/playwright:v1.25.0-jammy
- Operating System: Linux
- Node.js version: v14.20.0
- Browser: mobile-WebKit
- Extra: issue only happens when run in container with mobile-webkit devices; any other combination of set-ups don't have this issue. So far only notice this issue when going to [this page](https://secure.indeed.com/).

You can clone [this repo](https://github.com/Qibaoooo/playwright-issue) and run the test in playwright container to reproduce error.

**Code Snippet**

https://github.com/Qibaoooo/playwright-issue/blob/main/tests/demo-test.spec.ts

**Describe the bug**

Page keeps resizing by itself and stop being responsive. See attachment for the recording.
[video.webm](https://user-images.githubusercontent.com/21255238/188798764-af010c69-8aa4-4561-8b54-46ffd9de2678.webm)

**Comment by cwilliams7** [comment]
Experiencing the same issue, mobile-webkit works fine on my Mac but when I run it in a container (1.26.0-focal or jammy) screenshot comparisons stop working. I looked at the video playback and I see that the screen is constantly re-sizing which I think is causing the screenshot to fail because it is waiting for the page to be stable.

**Comment by sylvaingi** [comment]
Also observing this error on v1.28.0. The viewport is alternating very quickly between its intended width + height and the same dimensions divided by two.

Tried fiddling with the `deviceScaleFactor` using `1` as a value to disable high dpr emulation, it did not work.

**Comment by daniel-meneses** [comment]
+1 as described using mcr.microsoft.com/playwright:v1.27.0-focal

**Comment by deviantintegral** [comment]
Still seeing this in 1.31.1, not using the docker container (but, running in Docker). Anyone have a workaround for CI scenarios?

**Comment by chadjohnson21** [comment]
Still seeing this on iPhone (all models) device profile on Playwright 1.37.0 running on Debian 11 bullseye

**Comment by nmerget** [comment]
I saw this for Playwright 1.43.1 (Windows and mcr.microsoft.com/playwright:v1.43.1).

In my special case one part in my component caused the issue:

` document.documentElement.style.blockSize = "100%";
      document.documentElement.style.overflow = "hidden";`
      
I used the code to force some fixed layout for the page. And it works as expected for all browsers except for `mobile-webkit`.

We've also had the [luxury](https://github.com/db-ui/mono/actions/runs/8779792401/job/24088580438) of testing it for Angular, React and Vue and all failed, so it's not a framework-specific thing.
My workaround was to opt out the execution and add the desired styles via `css`, but just for the tests.

For this [page](https://secure.indeed.com/) is there some `useEffect` changing the style of the `document`? Maybe this is causing the rerender of the entire page?

To summarize, I guess it's a bug in mobile-webkit maybe related to the `page` object provided by the playwright `test`. It's kind of wired that all other projects and even dektop-webkit works as expected...

**Comment by kasper573** [comment]
Is this related to or possibly the same issue as https://github.com/microsoft/playwright/issues/21784 ?

**Comment by dyllandry** [comment]
Observing on mcr.microsoft.com/playwright:v1.45.1-jammy


Is there any workaround? I don't know if I'd be able to narrow it down to a specific line of css, unless someone can offer advice there. For now I'll disable iPhone tests.

**Comment by dyllandry** [comment]
I was able to get my tests working by running playwright in headed mode and using `xvfb-run`!

```
xvfb-run npx playwright test --headed --config playwright.iphone.ts
```

Because these headed tests are a lot slower, I separated out all the iPhone projects/devices into a separate playwright configuration from all the chrome/firefox/safari tests so that I can run just the iPhone tests this way. For the remaining tests I don't use xvfb-run or headed mode to keep them fast.

I don't remember where I got this work around from, I don't think I came up with it on my own.

**Comment by harsh-sql** [comment]
@dyllandry I am also doing that only. I have also made a separate project and defined headless as false and it works fine for now but yeah, it is slow so still finding a solution for that. 
I tried to explicitly define viewport but that does not work I guess because headless is all together a separate browser and have different settings that cannot be changed in config file so leaving it as it is until any update comes for such a thing.

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your contribution to our project. This issue has been closed due to its limited upvotes and recent activity, and insufficient feedback for us to effectively act upon. Our priority is to focus on bugs that reflect higher user engagement and have actionable feedback, to ensure our bug database stays manageable.

Should you feel this closure was in error, please create a new issue and reference this one. We're open to revisiting it given increased support or additional clarity. Your understanding and cooperation are greatly appreciated.
